pkg/tracing/tracing.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package tracing provides OpenTelemetry tracing infrastructure for the gateway-api-inference-extension
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	ServiceName = "gateway-api-inference-extension"
+
+	envOTELTracingEnabled      = "OTEL_TRACING_ENABLED"
+	envOTELExporterEndpoint    = "OTEL_EXPORTER_OTLP_ENDPOINT"
+	envOTELServiceName         = "OTEL_SERVICE_NAME"
+	envOTELSamplingRate        = "OTEL_SAMPLING_RATE"
+
+	OperationGatewayRequest       = "gateway.ext_proc.request"
+	OperationRequestOrchestration = "gateway.request_orchestration"
+	OperationScheduling           = "gateway.scheduling"
+
+	AttrGenAIRequestModel = "gen_ai.request.model"
+	AttrOperationOutcome  = "operation.outcome"
+
+	OutcomeSuccess = "success"
+	OutcomeError   = "error"
+)
+
+type Config struct {
+	Enabled          bool
+	ExporterEndpoint string
+	SamplingRate     float64
+	ServiceName      string
+}
+
+func NewConfigFromEnv() *Config {
+	config := &Config{
+		Enabled:          false,
+		ExporterEndpoint: "http://localhost:4317",
+		SamplingRate:     0.1,
+		ServiceName:      ServiceName,
+	}
+
+	if enabled := os.Getenv(envOTELTracingEnabled); enabled != "" {
+		if enabledBool, err := strconv.ParseBool(enabled); err == nil {
+			config.Enabled = enabledBool
+		}
+	}
+
+	if endpoint := os.Getenv(envOTELExporterEndpoint); endpoint != "" {
+		config.ExporterEndpoint = endpoint
+	}
+
+	if serviceName := os.Getenv(envOTELServiceName); serviceName != "" {
+		config.ServiceName = serviceName
+	}
+
+	if samplingRate := os.Getenv(envOTELSamplingRate); samplingRate != "" {
+		if rate, err := strconv.ParseFloat(samplingRate, 64); err == nil {
```

Comment:
Does it make sense to inform the user e.g. by failing during the creation whenever a value is not recognized?